### ISSUE TITLE
Lookahead + new Interpreter issue on RelayClassicMutation mutations

### ISSUE THREAD

**Issue by newx** [rmosolgo/graphql-ruby#2606]
Graphql-ruby version: 1.9.15
Ruby version: ruby 2.6.2p47

Currently when I try to use `lookahead` via `extras: [:lookahead]` on a mutation that inherits from `GraphQL::Schema::RelayClassicMutation` I get the following error:

```
resolve': missing keyword: lookahead (ArgumentError)
```

**How to reproduce it**

The code below works fine if you remove the `use(GraphQL::Execution::Interpreter)`

```ruby
require "bundler/inline"

gemfile do
  source 'https://rubygems.org'
  # gem "graphql", "1.9.15"
  gem "pry"
  gem "graphql", path: "~/Work/ActionBuilder/repos/graphql-ruby"
end

puts GraphQL::VERSION

class BaseField < GraphQL::Schema::Field
  # def resolve_field(obj, args, ctx)
  #   resolve(obj, args, ctx)
  # end
end

class BaseObject < GraphQL::Schema::Object
  field_class BaseField
end

class BaseInputObject < GraphQL::Schema::InputObject
end

class UpdateName < GraphQL::Schema::RelayClassicMutation
  input_object_class BaseInputObject

  field :output, String, null: false
  argument :name, String, required: true

  # NOTE: this "hack" works because it injects the `extras` class method return
  # values used here https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/schema/relay_classic_mutation.rb#L38
  # def self.extras
  #   [:lookahead]
  # end

  def resolve(name:, lookahead:)
    { output: "New name: #{name} - lookahead: #{lookahead.class.name}" }
  end
end

class MutationType < BaseObject
  field :update_name, mutation: UpdateName, null: false, extras: [:lookahead]
end

class Schema < GraphQL::Schema
  mutation(MutationType)
  use(GraphQL::Analysis::AST)
  # NOTE: this works fine on graphql-ruby master if you remove the Interpreter below.
  use(GraphQL::Execution::Interpreter)
end

p Schema.execute('mutation { updateName(input: { name: "John Doe"}) { output } }')
```

**Results**

Without the new Interpreter:

```ruby
#<GraphQL::Query::Result @query=... @to_h={"data"=>{"updateName"=>{"output"=>"New name: John Doe - lookahead: GraphQL::Execution::Lookahead"}}}>
```

With the new Interpreter (`use(GraphQL::Execution::Interpreter)`):

```ruby
#=> mutation_lookahead_test.rb:38:in `resolve': missing keyword: lookahead (ArgumentError)
```

This PR fixes this issue for me https://github.com/rmosolgo/graphql-ruby/pull/2605
because I'm only using lookahead on `extras` argument.

cc @rmosolgo

**Comment by rmosolgo** [comment]
Hi, thanks for the detailed report here! It looks like we're not checking properly for configured `extras`. You can see here the code that is _supposed_ to cover this case: 

https://github.com/rmosolgo/graphql-ruby/blob/9d2bac94cc2c4494b2201d4356306ae26029e8f8/lib/graphql/schema/relay_classic_mutation.rb#L38-L44

However, it only checks `class.extras`, which are the extras defined _in the mutation class_, eg 

```ruby 
class UpdateName < GraphQL::Schema::RelayClassicMutation
  extras [:lookahead]
end 
```

But it doesn't see the ones added to the field definition, as in your example `field ... extras: [...]`. 

So, I think that code should be improved to (somehow) check the field-defined extras. In fact, I can't think of a way to implement this behavior out of the box. Resolvers don't have a reference to the field instance which is calling them, so the first step of the fix is to pass in a `field:` option here: 

https://github.com/rmosolgo/graphql-ruby/blob/9d2bac94cc2c4494b2201d4356306ae26029e8f8/lib/graphql/schema/resolver.rb#L32

Then improve the `.extras.each` code above to also check field-defined extras. 

As a work-around, you could move the lookahead configuration into the mutation class, which will make it work!

**Comment by newx** [comment]
Thanks for the feedback @rmosolgo. I will attempt make these changes and fix this.

**Comment by rmosolgo** [comment]
Fixed by #2605
